File: Java_Fundamentals_January_2017/Java_Advanced/Exersices/03. Stacks/src/StackFibonacci.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;

public class StackFibonacci {
    public static void main(String[] args) throws IOException {
        BufferedReader in =
                new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(in.readLine());

        ArrayDeque<Long> fibNumbers = new ArrayDeque<>();
        fibNumbers.push((long)1);
        fibNumbers.push((long)1);

        while (fibNumbers.size() <= n){
            long f0 = fibNumbers.pop();
            long f1 = fibNumbers.peek();

            fibNumbers.push(f0);
            fibNumbers.push(f0 + f1);
        }

        System.out.println(fibNumbers.peek());
    }
}

File: Java_Fundamentals_January_2017/Java_Advanced/Exersices/02. Matrices C#/Matrices/Stack Fibonacci/StackFibonacci.cs
namespace Stack_Fibonacci
{
    using System;
    using System.Collections.Generic;

    public class StackFibonacci
    {
        public static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());

            Stack<long> fibNumbers = new Stack<long>();
            
            fibNumbers.Push(1);
            fibNumbers.Push(1);

            while (fibNumbers.Count <= n)
            {
                long f0 = fibNumbers.Pop();
                long f1 = fibNumbers.Peek();

                fibNumbers.Push(f0);
                fibNumbers.Push(f0 + f1);
            }

            Console.WriteLine(fibNumbers.Peek());
        }
    }
}
